### PR TITLE
Settings re-organised and fixed issue with tag overwritten on STB

### PR DESCRIPTION
**Pull request by phunkyfish**

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -185,7 +185,7 @@ void ADDON_ReadSettings(void)
   else
     g_strTimeshiftBufferPath = DEFAULT_TSBUFFERPATH;
 
-  // read setting "enabletimeshift" from settings.xml 
+  // read setting "readtimeout" from settings.xml
```
Comment (ksooo) [inline src/client.cpp]: I'm not a big fan of comments just stating what is obvious.

File: src/Timers.cpp [hunk 2]
```diff
@@ -173,20 +174,45 @@ std::vector<Timer> Timers::LoadTimers()
...
+    {
+        XBMC->Log(LOG_NOTICE, "RNL No, %s %s", timer.strTitle.c_str(), timer.tags.c_str());
+    }
+
+    if (findTagInTimerTags("Manual", timer.tags))
+    {
+      //We create a Manual tag on Manual timers created from Kodi, this allows us to set the Timer Type correctly
+      if (timer.iWeekdays != PVR_WEEKDAY_NONE)
+      {
+          XBMC->Log(LOG_NOTICE, "RNL MAN REPEAT");
```
Comment (ksooo) [inline src/Timers.cpp]: indentation

File: src/Timers.cpp [hunk 3]
```diff
@@ -173,20 +174,45 @@ std::vector<Timer> Timers::LoadTimers()
...
 
-    if (tags == "Manual")
+    if (findTagInTimerTags("Manual", timer.tags))
     {
-      if (timer.iWeekdays)
+        XBMC->Log(LOG_NOTICE, "RNL Yes, %s %s", timer.strTitle.c_str(), timer.tags.c_str());
+    }
+    else
+    {
+        XBMC->Log(LOG_NOTICE, "RNL No, %s %s", timer.strTitle.c_str(), timer.tags.c_str());
```
Comment (ksooo) [inline src/Timers.cpp]: indentation

File: src/Timers.cpp [hunk 4]
```diff
@@ -173,20 +174,45 @@ std::vector<Timer> Timers::LoadTimers()
...
+      }
     }
     else
-    { //Default to EPG
-      timer.type = Timer::EPG_ONCE;
+    { //Default to EPG for all other standard timers
+      if (timer.iWeekdays != PVR_WEEKDAY_NONE)
+      {
+          XBMC->Log(LOG_NOTICE, "RNL EPG REPEAT");
+          timer.type = Timer::EPG_REPEATING;
```
Comment (ksooo) [inline src/Timers.cpp]: intentation

File: src/Timers.cpp [hunk 5]
```diff
@@ -173,20 +174,45 @@ std::vector<Timer> Timers::LoadTimers()
...
+      }
+      else
+      {
+          XBMC->Log(LOG_NOTICE, "RNL EPG ONCE");
```
Comment (ksooo) [inline src/Timers.cpp]: once again, indentation

File: src/Timers.cpp [hunk 6]
```diff
@@ -198,6 +224,13 @@ std::vector<Timer> Timers::LoadTimers()
   return timers; 
 }
 
+bool Timers::findTagInTimerTags(std::string tag, std::string tags)
```
Comment (ksooo) [inline src/Timers.cpp]: You should decide wheter you want to start method names with a smal or a capital letter. Currently, there is both.

File: src/Timers.h [hunk 7]
```diff
@@ -102,6 +104,7 @@ class Timers
 
   // functions
   std::vector<Timer> LoadTimers();
+  bool findTagInTimerTags(std::string tag, std::string tags);
```
Comment (ksooo) [inline src/Timers.h]: Looking at the implementation, this method could be `static`.

File: src/client.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/client.cpp]: Done

File: src/Timers.cpp (hunk 2 above)
Comment (phunkyfish) [reply src/Timers.cpp]: Done

File: src/Timers.cpp (hunk 4 above)
Comment (phunkyfish) [reply src/Timers.cpp]: Done

File: src/Timers.cpp (hunk 6 above)
Comment (phunkyfish) [reply src/Timers.cpp]: Done

File: src/Timers.h (hunk 7 above)
Comment (phunkyfish) [reply src/Timers.h]: Done

**Comment by phunkyfish** [comment]
All changed incorporated. Should be ready to go.

**Comment by phunkyfish** [comment]
Not related to this PR but I have a question on my autotimers implementation. If there is a better forum to ask these questions please let me know.

When I create an autotimer from the EPG it defaults to Any Channels instead of the channel I was on in the EPG. Am I missing something to have it select the current channel or is this expected behaviour. I would still like to be able to select any channels just not as the default.

And FYI, I’ve been runnnig the above PR for several days without issue so it should be fine to merge.

**Comment by ksooo** [comment]
>  Not related to this PR but I have a question on my  autotimers implementation. If there is a better forum to ask these  questions please let me know.

 You could send my PM on forum.kodi.tv

> When I create an autotimer from the EPG it defaults to Any Channels instead of the channel I was on in the EPG. Am I missing something to have it select the current channel or is this expected behaviour. I would still like to be able to select any channels just not as the default.

 It's some time ago I implemented this, but AFAIR this is currently not possible, sorry.

> And FYI, I’ve been runnnig the above PR for several days without issue so it should be fine to merge.

 Ooops. I simply forgot to merge. Thanks for the heads-up.

**Comment by ksooo** [comment]
I don't think this should be default behavior. The default values for an auto timer should be very relaxed - any time, any day, any channel. I just want to record the shows and I don't care for the actual channel, the time, etc.

**Comment by phunkyfish** [comment]
Fair point. I guess if the duplicate detection is good enough this should work well. 

It will be interesting to see how SD vs HD works when I finish the implementation as there are a few options. I can use a setting to mark quality preference and go by that I guess.

**Pull request merged**